app/actions/websocket/channel.ts

```diff
@@ -1,21 +1,68 @@
+import {localRemoveCurrentUserFromChannel, localSetChannelDeleteAt, switchToChannel} from '@actions/local/channel';
+import {fetchChannelInfo, fetchMyChannel} from '@actions/remote/channel';
+import {fetchUsersByIds, updateNewUsersVisible, updateUsersNoLongerVisible} from '@actions/remote/user';
 import Events from '@constants/events';
 import DatabaseManager from '@database/manager';
 import {queryActiveServer} from '@queries/app/servers';
 import {deleteChannelMembership, queryCurrentChannel} from '@queries/servers/channel';
-import {queryConfig, setCurrentChannelId} from '@queries/servers/system';
+import {queryCommonSystemValues, queryConfig, setCurrentChannelId} from '@queries/servers/system';
 import {queryLastChannelFromTeam} from '@queries/servers/team';
-import {queryCurrentUser} from '@queries/servers/user';
+import {queryCurrentUser, queryUserById} from '@queries/servers/user';
 import {dismissAllModals, popToRoot} from '@screens/navigation';
 import {isTablet} from '@utils/helpers';
 import {isGuest} from '@utils/user';
 
+export async function handleUserAddedEvent(serverUrl: string, msg: any) {
+    const database = DatabaseManager.serverDatabases[serverUrl];
+    if (!database) {
+        return;
+    }
+    const {currentChannelId, currentUserId} = await queryCommonSystemValues(database.database);
+    const teamId = msg.data.team_id;
+    const channelId = msg.data.channel_id;
+    const userId = msg.data.user_id;
+
+    const models: Model[] = [];
+
+    try {
+        // Handle new visible users for guests
+        const addedUser = queryUserById(database.database, userId);
+        if (!addedUser) {
+            const {users} = await fetchUsersByIds(serverUrl, [userId], true);
+            if (users) {
+                models.push(...await database.operator.handleUsers({users, prepareRecordsOnly: true}));
+            }
+        }
+
+        if (userId === currentUserId) {
+            const {channels, memberships} = await fetchMyChannel(serverUrl, teamId, channelId, true);
+            if (channels && memberships) {
+                models.push(...await database.operator.handleMyChannel({channels, myChannels: memberships, prepareRecordsOnly: true}));
+            }
+            const {users} = await updateNewUsersVisible(serverUrl, true);
+            if (users) {
+                models.push(...await database.operator.handleUsers({users, prepareRecordsOnly: true}));
+            }
+        } else if (msg.broadcast.channel_id === currentChannelId) {
+            const {channelInfo} = await fetchChannelInfo(serverUrl, channelId, true);
+            if (channelInfo) {
+                models.push(...await database.operator.handleChannelInfo({channelInfos: [channelInfo], prepareRecordsOnly: true}));
+                models.push(...await database.operator.handleChannelMembership({channelMemberships: [{channel_id: channelId, user_id: userId}], prepareRecordsOnly: true}));
+            }
+        }
```

Comment:
why do we need this too ?

Reply:
If a user gets added to a channel, we should update that membership. Not sure how often it is refreshed or where it is queried, but I guess we should keep the information up to date.

Also, the channelInfo for that channel has almost sure changed (the member count has increased one), so we update that. We could add the count manually (memberCount += 1), but I preferred to go to the server to be on the "safe side".

Reply:
Lol ok I agree with the sentimment, now the question is where in the Ui we display that info? Channel info screen? Do we need the dtats to show somewhere else? Can we update it only when we access that screen?

Just trying to be on the "safe other side" hahhaha

If you want we can have a chat as we seem to be going back and forth on this PR